src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -1005,6 +1032,18 @@ void closeStream(int streamid) {
         if (finalStream() && streams.isEmpty()) {
             // should be only 1 stream, but there might be more if server push
             close();
+        } else {
+            // Start timer if property present and not already created
+            synchronized (this) {
```

Comment:
I think we can avoid this `synchronized` block every time we close a non-final stream, by slightly changing this code to something like:

```
final Optional<Duration> idleTimeoutDuration = client().idleConnectionTimeout();
if (idleTimeoutDuration.isPresent()) {
  synchronized(this) {
    // idleConnectionTimerEvent is always accessed within a synchronized block
    if (streams.isEmpty() && idleConnectionTimeoutEvent == null) {
      idleConnectionTimeoutEvent = idleTimeoutDuration.map(IdleConnectionTimeoutEvent::new).get();
      client().registerTimer(idleConnectionTimeoutEvent);
    }
  }
}
```
This would avoid the `synchronized` block when no idle timeout is configured.